aea/cli/generate_all_protocols.py

```diff
@@ -257,7 +257,9 @@ def _check_preliminaries() -> None:
     )
 
 
-def download_package(package_id: PackageId, destination_path: str) -> None:
+def download_package(
+    package_id: PackageId, destination_path: str
+) -> None:  # pragma: nocover
```

Comment:
We cannot use this method since the remote HTTP registry is not working right now and we can just patch this method to test the main logic

Reply:
I am vehemently against the use of `# pragma: no cover`. Just don't cover it then but don't hide it.
```
try:
    ...
except ConnectionError as e:
    logging.error("Could not download {package_id}:\n{e}")
```
this way test coverage will show it was not covered on a particular run and it can be found in the logs.

Reply:
Not using `# pragma: nocover` is not gonna change anything and it might also confuse the next person who will be working on the test since there won't be any test coverage for the method and they'll try to add test for this. Having this mark will at least them know that this was intentional, we don't want to have coverage for this method

Reply:
but eventually we do want to have coverage for this method, just cannot right now. The fact that remote HTTP registry is not working being the reason is also not clear from the tag. I urge you to not use `# pragma no cover` as it hides problems and we have had bad experiences with this in the past. I rather have a comment explaining why it is not covered than hiding it, then coverage will be 99% instead of 100% and that is fine.